src/tools/perf/perftest.c

```diff
@@ -822,26 +809,63 @@ static void ctx_free(struct perftest_context *ctx)
     free(ctx->params.super.msg_size_list);
 }
 
-static ucs_status_t setup_rte(struct perftest_context *ctx)
+static const perftest_rte_t plugin_sock_rte = {
+    .setup   = setup_sock_rte,
+    .cleanup = cleanup_sock_rte
+};
+
+const perftest_rte_t *rte_list[PERFTEST_RTE_MAX] = {
+    [PERFTEST_RTE_SOCK] = &plugin_sock_rte,
+#if defined (HAVE_MPI)
+    [PERFTEST_RTE_MPI] = &plugin_mpi_rte,
+#endif
+};
+
+static const perftest_rte_t *rte;
+
+static int lookup_rte(struct perftest_context *ctx)
 {
     if (ctx->mad_port != NULL) {
-        return setup_mad_rte(ctx);
+        return PERFTEST_RTE_MAD;
     } else if (ctx->mpi) {
-        return setup_mpi_rte(ctx);
+        return PERFTEST_RTE_MPI;
     } else {
-        return setup_sock_rte(ctx);
+        return PERFTEST_RTE_SOCK;
     }
 }
 
-static void cleanup_rte(struct perftest_context *ctx)
+static const char *perftest_rte_str(perftest_rte_type_t type)
 {
-    if (ctx->mad_port != NULL) {
-        cleanup_mad_rte(ctx);
-    } else if (ctx->mpi) {
-        cleanup_mpi_rte(ctx);
-    } else {
-        cleanup_sock_rte(ctx);
+    switch (type) {
+    case PERFTEST_RTE_SOCK:
```

Comment:
can we make this a field of the RTE struct?

Reply:
added to new structure